yun/libraries/Parse/src/internal/ParseClient.h

```diff
@@ -127,13 +127,14 @@ class ParseClient {
    *  \param   httpVerb - GET/DELETE/PUT/POST
    *  \param   httpPath - the endpoint of REST API e.g. /1/installations
    *  \param   requestBody - http request body in json format, leave it as "" for DELTE/GET requests
-   *  \param   urlParams - leave it as "" unless to perform a Parse query,
-   *            use it to specify query condition e.g. where={"KEY1"":VALUE1}&limit=10&keys=KEY1,KEY2
-   *  \result response of request
+   *           or use it to specify query condition e.g. where={"KEY1"":VALUE1}&limit=10&keys=KEY1,KEY2
+   *           NOTE: isQuery needs to be set as true if you use requestBody as query condition
+   *  \param   isQuery - if the request is a query
+   *  \result  response of request
    *  NOTE: ParseObjectCreate, ParseObjectUpdate, ParseObjectDelete, ParseGETObject,
    *  ParseQuery, ParseCloudFunction, ParseTrackEvent can also be used for REST API call
    */
-  ParseResponse sendRequest(const char* httpVerb, const char* httpPath, const char* requestBody, const char* urlParams);
+  ParseResponse sendRequest(const char* httpVerb, const char* httpPath, const char* requestBody, bool isQuery);
```

Comment:
1. This would be a breaking change in the public API surface of the SDK.  Customers that already use this method can't upgrade the SDK without changing their code.  We should avoid doing breaking changes like this.
2. Overloading requestBody to mean 2 things could be confusing. Most people think of HTTP request body as the content that's after HTTP headers, but the query string is actually in the URL.

I'd recommend not changing the public API, and fix this issue internally inside the SDK implementation.